contracts/protocol/facets/OfferHandlerFacet.sol

```diff
@@ -402,6 +409,30 @@ contract OfferHandlerFacet is IBosonOfferHandler, OfferBase {
         emit OfferRoyaltyInfoUpdated(_offerId, offer.sellerId, _royaltyInfo, msgSender());
     }
 
+    /**
+     * @notice Internal function to update the mutualizer address for an offer.
+     *
+     * Emits an OfferMutualizerUpdated event if successful.
+     *
+     * Reverts if:
+     * - The offers region of protocol is paused
+     * - Offer does not exist
+     * - Caller is not the assistant of the offer
+     * - Offer has already been voided
+     *
+     * @param _offerId - the id of the offer to update
+     * @param _newMutualizer - the new mutualizer address (can be zero for self-mutualization)
+     */
+    function updateOfferMutualizerInternal(uint256 _offerId, address _newMutualizer) internal {
+        // Make sure the caller is the assistant, offer exists and is not voided
+        Offer storage offer = getValidOfferWithSellerCheck(_offerId);
+
+        DisputeResolutionTerms storage disputeResolutionTerms = fetchDisputeResolutionTerms(_offerId);
+        disputeResolutionTerms.mutualizerAddress = payable(_newMutualizer);
```

Comment:
Make sure here that old mutualizer is not the same as the new one, to prevent no-op transactions.